src/Symfony/Component/Security/Http/EventListener/CsrfProtectionListener.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Security\Core\Exception\InvalidCsrfTokenException;
+use Symfony\Component\Security\Csrf\CsrfToken;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+use Symfony\Component\Security\Http\Authenticator\CsrfProtectedAuthenticatorInterface;
+use Symfony\Component\Security\Http\Event\VerifyAuthenticatorCredentialsEvent;
+
+class CsrfProtectionListener implements EventSubscriberInterface
+{
+    private $csrfTokenManager;
+
+    public function __construct(CsrfTokenManagerInterface $csrfTokenManager)
+    {
+        $this->csrfTokenManager = $csrfTokenManager;
+    }
+
+    public function verifyCredentials(VerifyAuthenticatorCredentialsEvent $event): void
+    {
+        $authenticator = $event->getAuthenticator();
+        if (!$authenticator instanceof CsrfProtectedAuthenticatorInterface) {
+            return;
+        }
+
+        $csrfTokenValue = $authenticator->getCsrfToken($event->getCredentials());
+        if (null === $csrfTokenValue) {
```

Comment:
Hmm, this might be a problem. It would make it easy for someone to write an authenticator where a bad user could simply "omit" the `_token` field in order to skip CSRF checking. If we need an "opt-out" from this interface, it needs to be a different method - `shouldCheckCsrfProtection()` - maybe with a `Request` argument just in case. Then, if that's true, ALWAYS validate.

Reply:
marking as resolved, the badge system (https://github.com/symfony/symfony/pull/33558#issuecomment-610111667) will remove this interface